serokell-style.md

```diff
@@ -93,10 +102,16 @@ as constructor name (also applies to `newtype`).
 data User = User Int String
 ```
 
-Field name for `newtype` should start with `get` prefix followed by type name.
+Field name for `newtype` should start with `un` or `run` prefix followed by type name.
+
+* `run` for wrappers with monadic semantic.
```

Comment:
Seems better to remain `run` for smart runner and something else for getter.
E.g
```
newtype MyMonad m a = MyMonad { unMyMonad :: ReaderT Int m a }

runMyMonad :: Int -> MyMonad m a -> m a
```

Reply:
@Martoon-00 Well, actually I want to be consistent with `universum` and `transformers` here. You see, for `ReaderT`:

```haskell
newtype ReaderT r m a = ReaderT { runReaderT :: r -> m a }
```

function `runReaderT` has type `runReaderT :: ReaderT r m a -> r -> m a`. Monad is the first argument. Same with `MyMonad`:

```haskell
newtype MyMonad m a = MyMonad { runMyMonad :: ReaderT Int m a }
```

```
runMyMonad :: MyMonad m a -> ReaderT Int m a
```
`MyMonad` is still the first argument. I can add to this proposal that smart runners should have name started with `using`. So, your example in new style-guide:

```haskell
newtype MyMonadT m a = MyMonadT { runMyMonadT :: ReaderT Int m a }

usingMyMonad :: Int -> MyMonad m a -> m a
usingMyMonad int = usingReaderT int . runMyMonadT
```

Reply:
Usually when you write smart monadic unwrappers, it's clear from context, where you do `newtype` unwrapping. I prefer consistency with `transformers` and `mtl` rather than uniformity across every `newtype`. But that's subjective.